.github/workflows/main.yml

```diff
@@ -0,0 +1,33 @@
+name: Integration Test
+
+on:
+  push:
+
+jobs:
+  main:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build emissions-api
+        run: docker-compose -f .test/docker-compose.yml build emissionsapi
+      - name: Spin up database and emissions-api web server
+        run: docker-compose -f .test/docker-compose.yml up -d
+      - name: Wait for database
+        run: until PGPASSWORD=user psql -h 127.0.0.1 -U user db -c '\q'; do sleep 1; done
+      - name: Check if emissions-api web server is up
+        run: nc -z 127.0.0.1 8000
+      - name: Download and include data to the emissions-api service
+        run: docker-compose -f .test/docker-compose.yml exec -T emissionsapi /opt/emissions-api/venv/bin/emissionsapi-autoupdater
+      - name: Products API
+        run: 'curl --fail -X GET "http://localhost:8000/api/v2/products.json" -H  "accept: application/json"'
+      - name: Average API
+        run: 'curl --fail -X GET "http://localhost:8000/api/v2/methane/average.json" -H  "accept: application/json"'
+      - name: Data Range API
+        run: 'curl --fail -X GET "http://localhost:8000/api/v2/carbonmonoxide/data-range.json" -H  "accept: application/json"'
+      - name: Geo API
+        run: 'curl --fail -X GET "http://localhost:8000/api/v2/carbonmonoxide/geo.json" -H  "accept: application/json"'
+      - name: Statistics API
+        run: 'curl --fail -X GET "http://localhost:8000/api/v2/carbonmonoxide/statistics.json" -H  "accept: application/json"'
+      - name: Show logs
+        if: ${{ failure() }}
+        run: docker-compose -f .test/docker-compose.yml logs
```

Comment:
Did you try this? Does it work?
I remember looking for something like this once but GitHub Actions would halt the pipeline if an earlier test failed.
But maybe that has changed since then.

Reply:
Yes, Not with the latest version, but I used it for debugging. See https://github.com/shaardie/emissions-api/runs/1626894288?check_suite_focus=true